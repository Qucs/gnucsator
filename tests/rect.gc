load bm_trivial.so
load s_sparam.so
load c_hide.so
load d_subckt.so
include ../include/sources.v
verilog
hidemodule *

param u=3
param foo=TH+TL+Tr+Tf

Vrect #(.TH(1) .TL(2) .Tr(0.1) .Tf(0.1) .U(u) .Td(1)) foo(1,0);

list

spice
.options noinsensitive
.print tran v(nodes)
.tran 0 10 10 trace=a

.clear
.unload c_hide.so

.end
