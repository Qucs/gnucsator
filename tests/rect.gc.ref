R: already installed, replacing
stashing as R:0
C: already installed, replacing
stashing as C:0
Vrect #(.U(u),.Td(1),.Tr(0.1),.Tf(0.1),.TH(1),.TL(2)) foo (.p(1),.n(0));
#Time       v(1)      
 0.         0.        
 0.1        0.        
 1.         0.        
 1.1        3.        
 2.3        3.        
 2.4        0.        
 4.2        3.f       
 4.3        3.        
 5.5        3.        
 5.6        16.f      
 7.4        19.f      
 7.5        3.        
 8.7        3.        
 8.8        6.f       
 10.        8.f       
