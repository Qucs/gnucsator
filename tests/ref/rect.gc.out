// This File is part of gnucap-qucs 
// (C) 2018 Felix Salfelder 
// GPLv3+ 
// mapping qucsator names to actual devices 
// "sources" section 
//* Vexp:V1 _net4 _net5 U1="0 V" U2="1 V" T1="0" T2="1 ms" Tr="1  ms" Tf="1 ms" 
//* Ipulse:I2 _net2 _net3 I1="0" I2="1 A" T1="0" T2="1 ms" Tr="1 ms" Tf="2 ms" 
//* Vpulse:V2 _net2 gnd U1="0 V" U2="1 V" T1="5m" T2="10m" Tr="1 ns" Tf="1 ns" 
//* Irect:I1 _net0 _net1 I="1" TH="1 ms" TL="1 ms" Tr=".5m" Tf=".5m" Td=".5m" 
//* Vac:V1 Gate gnd U="5 V" f="10 MHz" Phase="0" Theta="0" 
//* don't need spice for this 
//* .subckt CCVS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* H1 2 3 R1 {G} 
//* .ends 
//* .subckt CCCS(1 2 3 4); 
//* .parameter G=1 
//* .R:0 R1 1 4 1n 
//* F1 2 3 R1 {G} 
//* .ends 
//* 
//* only admit in dc and tr 
Vrect #(.TH(1),.TL(2),.Td(1),.Tf(0.1),.Tr(0.1),.U(u)) foo (.p(1),.n(0));
#Time       v(1)      
 0.         0.        
 0.1        0.        
 1.         0.        
 1.1        3.        
 1.35       3.        
 1.6        3.        
 1.7667     3.        
 1.9333     3.        
 2.1        3.        
 2.2        0.        
 2.2001     0.        
 2.2003     0.        
 2.2003     0.        
 2.2004     0.        
 4.2        0.        
 4.3        3.        
 4.6333     3.        
 4.9667     3.        
 5.1333     3.        
 5.3        3.        
 5.4        0.        
 5.4001     0.        
 5.4003     0.        
 5.4003     0.        
 5.4004     0.        
 7.4        0.        
 7.5        3.        
 7.8333     3.        
 8.1667     3.        
 8.3333     3.        
 8.5        3.        
 8.6        0.        
 8.6001     0.        
 8.6003     0.        
 8.6003     0.        
 8.6004     0.        
 10.        0.        
